jina/serve/networking.py

```diff
@@ -920,22 +831,51 @@ def send_request_sync(
                     tls=tls,
                     root_certificates=root_certificates,
                 ) as channel:
-                    if type(request) == DataRequest:
-                        metadata = (('endpoint', endpoint),) if endpoint else None
-                        stub = jina_pb2_grpc.JinaSingleDataRequestRPCStub(channel)
-                        response, call = stub.process_single_data.with_call(
-                            request,
-                            timeout=timeout,
-                            metadata=metadata,
-                        )
-                    elif type(request) == ControlRequest:
-                        stub = jina_pb2_grpc.JinaControlRequestRPCStub(channel)
-                        response = stub.process_control(request, timeout=timeout)
+                    metadata = (('endpoint', endpoint),) if endpoint else None
+                    stub = jina_pb2_grpc.JinaSingleDataRequestRPCStub(channel)
+                    response, call = stub.process_single_data.with_call(
+                        request,
+                        timeout=timeout,
+                        metadata=metadata,
+                    )
                     return response
             except grpc.RpcError as e:
                 if e.code() != grpc.StatusCode.UNAVAILABLE or i == 2:
                     raise
 
+    @staticmethod
+    def send_health_check_sync(
+        target: str,
+        timeout=100.0,
+        tls=False,
+        root_certificates: Optional[str] = None,
+    ) -> health_pb2.HealthCheckResponse:
+        """
+        Sends a request synchronously to the target via grpc
+
+        :param target: where to send the request to, like 127.0.0.1:8080
+        :param timeout: timeout for the send
+        :param tls: if True, use tls encryption for the grpc channel
+        :param root_certificates: the path to the root certificates for tls, only used if tls is True
+
+        :returns: the response health check
+        """
+
+        for i in range(3):
+            try:
+                with GrpcConnectionPool.get_grpc_channel(
+                    target,
+                    tls=tls,
+                    root_certificates=root_certificates,
+                ) as channel:
+                    health_check_req = health_pb2.HealthCheckRequest()
+                    health_check_req.service = ''
+                    stub = health_pb2_grpc.HealthStub(channel)
+                    return stub.Check(health_check_req, timeout=timeout)
+            except grpc.RpcError as e:
+                if e.code() != grpc.StatusCode.UNAVAILABLE or i == 2:
+                    raise
```

Comment:
Do we need error handling here a la `InternalNetworkError`, reporting back to the client which target failed, that sort of stuff? Or will this be handled differently?

Reply:
here  u are sending the health check to a single entity, this check is not about sending to the complete Flow, that is in the other PR